### PR TITLE
Assessment of related resources of an evidence

### PR DESCRIPTION
**Pull request by oxisto**
This PR changes the way the assessment works. Currently, once an evidence arrived, it is immediately handled. This PR introduces the idea of "related resources". An evidence can specify a number of resource IDs that are related to it (next to the one that it represents). Typical use cases include a disk volume related to a virtual machine. The assessment will now wait until an evidence that represents this related resource is available to assess the original evidence. A new metric called `VirtualMachineDiskEncryptionEnabled` is included to demonstrate how this works in Rego.

### CODE REVIEW

File: service/assessment/assessment.go [hunk 1]
```diff
@@ -151,26 +232,83 @@ func (s *Service) AssessEvidence(_ context.Context, req *assessment.AssessEviden
...
+		err = s.handleEvidence(req.Evidence, resourceId, nil)
+
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "error while handling evidence: %v", err)
+		}
 
-	if err != nil {
 		res = &assessment.AssessEvidenceResponse{
-			Status: false,
+			Status: assessment.AssessEvidenceResponse_ASSESSED,
```
Comment (oxisto) [inline service/assessment/assessment.go:281]: This should not be ASSESSED but FAILED.

File: api/assessment/assessment.proto [hunk 2]
```diff
@@ -77,7 +85,22 @@ message ConfigureAssessmentResponse {}
 message TriggerAssessmentRequest { string someOption = 1; }
 
 message AssessEvidenceRequest { Evidence evidence = 1; }
-message AssessEvidenceResponse { bool status = 1; }
+message AssessEvidenceResponse {
+  enum AssessmentStatus {
+    UNKNOWN = 0;
+    WAITING_FOR_RELATED = 1;
+    ASSESSED = 2;
```
Comment (oxisto) [inline api/assessment/assessment.proto:92]: add FAILED

File: cli/commands/tool/tool_test.go [hunk 3]
```diff
@@ -61,8 +62,9 @@ func TestMain(m *testing.M) {
 	}
 	orchestratorService = service_orchestrator.NewService()
 
-	sock, server, _, err = service.StartDedicatedAuthServer(":0")
-	orchestrator.RegisterOrchestratorServer(server, orchestratorService)
+	sock, server, _, err = service.StartDedicatedAuthServer(":0", func(server *grpc.Server) {
```
Comment (oxisto) [inline cli/commands/tool/tool_test.go:65]: We could add a `WithOrchestrator` to the service package to make this a little be cleaner

File: service/auth.go [hunk 4]
```diff
@@ -162,8 +162,10 @@ func parseToken(token string, authConfig *AuthConfig) (jwt.Claims, error) {
 	return parsedToken.Claims, nil
 }
 
+type AdditionalStartFunc func(*grpc.Server)
+
 // StartDedicatedAuthServer starts a gRPC server containing just the auth service
-func StartDedicatedAuthServer(address string, opts ...service_auth.ServiceOption) (sock net.Listener, server *grpc.Server, authService *service_auth.Service, err error) {
+func StartDedicatedAuthServer(address string, add AdditionalStartFunc, opts ...service_auth.ServiceOption) (sock net.Listener, server *grpc.Server, authService *service_auth.Service, err error) {
```
[the remark this answers is not in the document]
Comment (lebogg) [reply service/auth.go]: Or would it even make sens to make it an ServiceOption as well? Calling `StartDedicatedAuthServer("someaddress", nil)` looks not so nice I think.

File: cli/commands/login/login_test.go [hunk 5]
```diff
@@ -48,7 +49,9 @@ func TestMain(m *testing.M) {
 	var (
 		err error
 	)
-	sock, server, _, err = service.StartDedicatedAuthServer(":0")
+
+	sock, server, _, err = service.StartDedicatedAuthServer(":0", func(*grpc.Server) {}, service_auth.WithApiKeySaveOnCreate(false))
```
Comment (lebogg) [inline cli/commands/login/login_test.go]: here, 2nd parameter is filled with empty func. In another case, it is set to nil. As mentioned in `auth.go`, consider to make it a service option

File: policies/bundles/TransportEncryptionAlgorithm/input.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (lebogg) [inline policies/bundles/TransportEncryptionAlgorithm/input.json:1]: This is for testing, right? I'm thinking about putting them in a dedicated test folder. What do you think?

**Review by lebogg — COMMENTED** [review]
Current sate of my review. As discussed in person, we found some bugs while testing it within our real CSP environment.

**Pull request closed without merging**